src/core/src/core_logic/PatchInstaller.py

```diff
@@ -291,7 +293,26 @@ def is_reboot_pending(self):
             pending_file_exists = os.path.isfile(self.REBOOT_PENDING_FILE_PATH)
             pending_processes_exists = self.package_manager.do_processes_require_restart()
             self.composite_logger.log_debug(" - Reboot required debug flags: " + str(pending_file_exists) + ", " + str(pending_processes_exists) + ".")
-            return pending_file_exists or pending_processes_exists
+            is_reboot_required = pending_file_exists or pending_processes_exists
+
+            # Use Ubuntu Pro Client api in case of Ubuntu VMs.
+            if self.env_layer.get_package_manager() != Constants.APT:
+                return is_reboot_required
+            else:
+                self.composite_logger.log_debug("package_manager is AptitudePackageManager.")
+                is_pro_api_success, pro_reboot_status = self.package_manager.is_reboot_pending()
+                self.composite_logger.log_debug("is_pro_api_success: {0}, reboot_status {1}".format(is_pro_api_success, pro_reboot_status))
+
+                # Ubuntu Pro Client api failed. return default status.
+                if not is_pro_api_success:
+                    return is_reboot_required
+
+                # Compare Ubuntu Pro Client result and default result. Log if there is mismatch.
+                if is_reboot_required != pro_reboot_status:
+                    self.composite_logger.log_debug("Reboot status mismatch. pro = {0}, default = {1}".format(pro_reboot_status, is_reboot_required))
+
+                # Ubuntu Pro Client result is considered truth even if there is mismatch.
+                return pro_reboot_status
```

Comment:
In general, we'd want to avoid PackageManager specific behaviors inside generalized functions like PatchInstaller or PatchAssessor where possible.

Also this comment applies in this instance: https://github.com/Azure/LinuxPatchExtension/pull/169/files#r1107385713

Reply:
@kjohn-msft : I see that your changes are still in review. Should i wait for that to be checked-in?

Reply:
It's sort of a predecessor to what you're doing here, so it'd help in more than one way if you looked at that PR - both in influencing and aligning what you do here, and getting that in. You can pre-emptively align before it merges once you get the picture.